lang/python.md

```diff
@@ -2,10 +2,155 @@
+When you execute a Python program that you’ve written, the interpreter reads the code and converts it into usable commands. The important thing to note is that all of this is done after the program has been executed.
+
+With a shell, the interpreting – or conversion – happens in real-time as you type the code into the computer or system. This means that the actual program is executing as you type. This gives you some idea of how your final code will look, and what your program is actually going to do.
+
+When you write code in a text file, none of that happens until you feed the document into an interpreter. If you have Python installed on your computer you can call upon the interpreter using a command line, but this step is done after you’ve already written the code.
+
+This makes it more difficult to spot errors in your code, and it can also be frustrating if the interpreter runs into issues, because they may not be as apparent as they would if you had used a shell. Still, a lot of developers prefer to use a text editing tool because it is simple and easy to do.
+
+There are text editors with increased functionality – like Notepad++ – which were specifically developed with programming in mind.
+
+### How to Get Started with Python?
+Before you can do anything with a programming language, you first need to configure the development environment. Now, we’re going to cover how to set up Python and the interpreter that will execute your custom programs.
+
+
+Python comes pre-installed on Mac and a majority of Linux distributions. However, you may need to download an updated version depending on how old your system is.
+
+**TIP:** You can easily check your Python version by opening the terminal and running the following command:
+
+```
+     python --version
+```
+If you’re running Windows, you’ll need to download Python from the Python Software Foundation (link below).
+
+If you need to download Python, the best place to go is the [official site](https://www.python.org/). You’ll need to download the version specific to your operating system and processor (32 or 64 bit).
+
+**Mac:** Most Mac OS X computers already have Python 2.x installed, which is perfectly fine to use. The best way to install the latest version is to use [Homebrew](https://brew.sh/) to activate and manage them. You’ll find instructions on [how to do that here](https://gist.github.com/mjhea0/3c7d97e1d20ab86eb41e). You can also download Python directly from the PSF if you prefer.
+
+**Linux:** Python is included with most distributions of Linux. Check your current version and be sure to upgrade using the package manager, if necessary.
+
+**Windows:** Just download Python from the PSF.
+
+**TIP:** If you’re using Windows, be sure to select the option that adds Python.exe to your system path during installation. You can do this by selecting the option next to “Add python.exe to path,” and then choosing your local hard drive as the install location.
+
+**Which Version Should I Use?**
```

Comment:
@AaquibAli: Need one empty line after this, just use `Enter`, so it will be fixed!
Same as the previous one! : )

Reply:
```suggestion
#### Which Version Should I Use?
```